pkg/storage/edv/documentprocessor/documentprocessor.go

```diff
@@ -7,12 +7,91 @@ SPDX-License-Identifier: Apache-2.0
 package documentprocessor
 
 import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jose"
 	"github.com/hyperledger/aries-framework-go/pkg/storage/edv"
 )
 
+const (
+	failMarshalStructuredDocument   = "failed to marshal structured document into bytes: %w"
+	failEncryptStructuredDocument   = "failed to encrypt structured document into JWE form: %w"
+	failJWESerialize                = "failed to serialize JWE: %w"
+	failDeserializeJWE              = "failed to deserialize JWE: %w"
+	failDecryptJWE                  = "failed to decrypt JWE: %w"
+	failUnmarshalStructuredDocument = "failed to unmarshal structured document: %w"
+)
+
+type marshalFunc func(interface{}) ([]byte, error)
+
 // DocumentProcessor represents a type that can encrypt and decrypt between
 // Structured Documents and Encrypted Documents.
 type DocumentProcessor interface {
 	Encrypt(*edv.StructuredDocument) (*edv.EncryptedDocument, error)
 	Decrypt(*edv.EncryptedDocument) (*edv.StructuredDocument, error)
 }
+
+// AriesDocumentProcessor uses Aries crypto to encrypt and decrypt between
+// Structured Documents and Encrypted Documents.
+type AriesDocumentProcessor struct {
```

Comment:
Shouldn't have the Aries prefix. Fixable if you don't put the interface and concrete implementation together.

Reply:
Fixed.